client.go

```diff
@@ -513,6 +515,52 @@ func (c *Client) GetServerConfig() (*Response, error) {
 	return c.baseGet(c.url(shared.APIVersion))
 }
 
+// GetMessageFromErr converts an error received by the client into a more
+// user-friendly message.  In addition to the prettified message, returns
+// a bool indicating whether or not LXD appears to be installed.
+func GetMessageFromErr(err error) (bool, string) {
+	msg := fmt.Sprintf(i18n.G("error: %v"), err)
+
+	t, ok := err.(*url.Error)
+	if !ok {
+		return false, msg
+	}
+
+	u, ok := t.Err.(*net.OpError)
+	if !ok {
+		return false, msg
+	}
+
+	if u.Op == "dial" && u.Net == "unix" {
+		var lxdErr error
+
+		sysErr, ok := u.Err.(*os.SyscallError)
+		if ok {
+			lxdErr = sysErr.Err
+		} else {
+			// Try a syscall.Errno as that is what's returned for CentOS
+			errno, ok := u.Err.(syscall.Errno)
+			if !ok {
+				return false, msg
+			}
+
+			lxdErr = errno
+			msg = fmt.Sprintf("%d %s", uintptr(errno), errno.Error())
+		}
+
+		switch lxdErr {
+		case syscall.ENOENT:
+			return false, i18n.G("LXD socket not found; is LXD installed and running?")
```

Comment:
On Thu, Jun 02, 2016 at 01:17:08PM -0700, Sean Christopherson wrote:
> > +		if ok {
> > +			lxdErr = sysErr.Err
> > +		} else {
> > +			// Try a syscall.Errno as that is what's returned for CentOS
> > +			errno, ok := u.Err.(syscall.Errno)
> > +			if !ok {
> > +				return false, msg
> > +			}
> > +
> > +			lxdErr = errno
> > +			msg = fmt.Sprintf("%d %s", uintptr(errno), errno.Error())
> > +		}
> > +
> > +		switch lxdErr {
> > +		case syscall.ENOENT:
> > +			return false, i18n.G("LXD socket not found; is LXD installed and running?")
> 
> Are you specifically looking to keep i18n dependencies out of client.go?

That, but also not translate error messages that could potentially be
emitted in other code (client.go is a "public" API that anyone can
use, not just an internal API for us).

> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/2075/files/07514e20454b826a7dc3c9924a884193d9502ff9#r65610640